### PR TITLE
Chat service init

### LINKED ISSUE #39: Init project & folder structure

### PR DESCRIPTION
**Pull request by GluterusMaximus**
Closes #39

### CODE REVIEW

File: server/microservices/chat/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  '/docs',
+  swaggerUi.serve,
+  swaggerUi.setup(undefined, {
+    swaggerOptions: {
+      url: '/swagger.json',
+    },
+  })
+);
+
+registerRoutes(app);
```
Comment (Secret333Boy) [inline server/microservices/chat/src/app.ts:26]: Use express router here

File: server/microservices/chat/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import db from './repository/sequelize';
+import app from './app';
+import { Server } from 'http';
+import { Socket } from 'net';
+const port = process.env.PORT || 3004;
+
+db.authenticate()
+  .then(() => {
+    console.log(
+      'Connection to auth database has been established successfully.'
```
Comment (Secret333Boy) [inline server/microservices/chat/src/index.ts:10]: Not auth database

File: server/microservices/chat/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        process.exit(1);
+      }, 10000);
+
+      connections.forEach((curr) => curr.end());
+      setTimeout(() => connections.forEach((curr) => curr.destroy()), 5000);
+    };
+
+    db.sync({ logging: false }).then(() => {
+      const server = app.listen(port, () => {
+        console.log(`Auth service is running at http://localhost:${port}`);
```
Comment (Secret333Boy) [inline server/microservices/chat/src/index.ts:32]: Not auth service

File: server/microservices/chat/src/routes/routes.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Express } from 'express';
+import ChatController from '../controllers/ChatController';
+
+const registerRoutes = (app: Express) => {
+  app.get('/ping', ChatController.ping);
+};
+
+export { registerRoutes };
```
Comment (Secret333Boy) [inline server/microservices/chat/src/routes/routes.ts:8]: Rename this file to index.ts and create router here

**Review by Secret333Boy — CHANGES_REQUESTED** [review]

File: server/microservices/chat/src/app.ts (hunk 1 above)
Comment (GluterusMaximus) [reply server/microservices/chat/src/app.ts]: Fixed in [62df0f0](https://github.com/Pigeons-Inc/pigeon/pull/55/commits/62df0f09c4001897b104deff0978dbf48b341a4d)

File: server/microservices/chat/src/index.ts (hunk 2 above)
Comment (GluterusMaximus) [reply server/microservices/chat/src/index.ts]: Fixed in [62df0f0](https://github.com/Pigeons-Inc/pigeon/pull/55/commits/62df0f09c4001897b104deff0978dbf48b341a4d)

File: server/microservices/chat/src/index.ts (hunk 3 above)
Comment (GluterusMaximus) [reply server/microservices/chat/src/index.ts]: Fixed in [62df0f0](https://github.com/Pigeons-Inc/pigeon/pull/55/commits/62df0f09c4001897b104deff0978dbf48b341a4d)

File: server/microservices/chat/src/routes/routes.ts (hunk 4 above)
Comment (GluterusMaximus) [reply server/microservices/chat/src/routes/routes.ts]: FIxed in [62df0f0](https://github.com/Pigeons-Inc/pigeon/pull/55/commits/62df0f09c4001897b104deff0978dbf48b341a4d)

**Review by Secret333Boy — APPROVED** [review]
Good

**Pull request merged**